### ISSUE TITLE
Split the logic out of the API files into a service layer- 4 endpoints

### ISSUE THREAD

**Issue by dleard** [bcgov/cas-registration#1061]
Blocked by #952 
Blocking #827 

This is part 2 of our API refactor. Part 1 was done in #952, where we split the single api files into separate files per endpoint. For this card we'll create a service layer, so that the routes themselves do not contain any logic & all the logic lives in the service layer.

user_operator, operator, operations, and user endpoints are already covered by other cards. For each of the following files:
- [x] api/business_structure
- [x] api/naics
- [x] api/regulated_products
- [x] api/reporting_activities

Do the following:
- [x] move all logic from the endpoints into service files. Probably most logic should go into bc_obps/service/data_access_service/[endpoint_name]_service.py. If the logic is more complicated than database access, you can make an intermediary service. See the developer-guide and the user_operator service layer for background info on how to split things
- [x]  refactor the endpoint tests into separate files to match how they're split up in registration/api
- [x] confirm that no performance issues have been introduced by the refactor--run the profiles

**Comment by BCerki** [comment]
Old description in case needed for reference: Blocked by #952 

This is part 2 of our API refactor. Part 1 was done in #952, where we split the single api files into separate files per HTTP method. For this card we'll create a service layer, so that the routes themselves do not contain any logic & all the logic lives in the service layer.

Steps:
- Create a new `service` directory under the `api` directory
- Create a directory for each endpoint object (user_operator, operator...)
- Create a file for each HTTP method (get, put, post, delete)
- Move the logic out of the API files into the service layer files & import as needed in the API files

The service directory structure should look similar to the API file structure (example, user_operator):
- api
  - service
    - user_operator
      - get.py
      - put.py
      - post.py
      - delete.py

For this card to be complete, there should be no logic in the API route files & all logic should now be in the service layer files & imported into the API route files.
